zmon-controller-app/src/main/java/org/zalando/zmon/service/impl/EventLogServiceImpl.java

```diff
@@ -0,0 +1,157 @@
+                                      @Nullable final Long fromMillis,
+                                      @Nullable final Long toMillis) {
+        final Optional<URI> baseQuery = getBaseQueryUri(limit, fromMillis, toMillis);
+        if (!baseQuery.isPresent()) {
+            return ImmutableList.of();
+        }
+
+        final String typesFilter = eventLogProperties.getCheckHistoryEventsFilter().stream()
+                .map(Object::toString)
+                .collect(Collectors.joining(","));
+
+        final URIBuilder queryBuilder = new URIBuilder(baseQuery.get())
+                .addParameter("types", typesFilter)
+                .addParameter("key", "checkId")
+                .addParameter("value", String.valueOf(checkDefinitionId));
+        try {
+            final URI query = queryBuilder.build();
+
+            return Arrays.asList(restOperations.getForObject(query, Event[].class));
+        } catch (URISyntaxException e) {
+            LOG.error("Failed to query for check events", e);
+        }
+        return ImmutableList.of();
+    }
+
+
+    private Optional<URI> getBaseQueryUri(final Integer limit, final Long fromMillis, final Long toMillis) {
+        final URIBuilder baseQueryBuilder;
+        try {
+            baseQueryBuilder = new URIBuilder(eventLogProperties.getUrl().toString());
+        } catch (URISyntaxException e) {
+            LOG.error("Invalid event log URI", e);
+            return Optional.empty();
+        }
+
+        if (limit != null) {
+            baseQueryBuilder.addParameter("limit", limit.toString());
+        }
+        if (fromMillis != null) {
+            baseQueryBuilder.addParameter("from", fromMillis.toString());
+        }
+        if (toMillis != null) {
+            baseQueryBuilder.addParameter("to", toMillis.toString());
+        }
+
+        try {
+            return Optional.of(baseQueryBuilder.build());
+        } catch (URISyntaxException e) {
+            LOG.error("Failed to build base query", e);
+        }
+
+        return Optional.empty();
+    }
+
+    private Event convert(EventlogEvent in) {
+        final String typeName = in.getTypeName();
+        final String flowId = in.getFlowId();
+
+        final Event e = new Event();
+        e.setTypeName(typeName);
+        e.setTime(in.getTime());
+        e.setFlowId(flowId);
+        e.setTypeId(in.getTypeId());
+
+        for (Map.Entry<String, JsonNode> ie : in.getAttributes().entrySet()) {
+            try {
+                e.setAttribute(ie.getKey(), mapper.writeValueAsString(ie.getValue()));
+            } catch (JsonProcessingException ex) {
+                LOG.error("Failed to convert event {} of type {}", flowId, typeName, ex);
+            }
+        }
+
+        return e;
+    }
+
+    private static class AlertEventsResponse extends LinkedList<EventlogEvent> {
+    }
+
+    private static class CheckEventsResponse extends LinkedList<Event> {
+    }
```

Comment:
were these left-overs? I believe I had the plan to use these for the getForObject() calls to avoid the array to list operation